pkg/blockbuilder/blockbuilder_metrics.go

```diff
@@ -0,0 +1,53 @@
+package blockbuilder
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type blockBuilderMetrics struct {
```

Comment:
Should we add a metric for number of partitions handled by this block builder? Other way to determine is taking the count on perPartitionLog in queries, so maybe it is not required